scrabble.js

```diff
@@ -1,8 +1,147 @@
-var Scrabble = function() {};
+var Scrabble = function(word) {
+  this.word = word;
+};
+
+var Scoring = {
+  scoreChart: {
+      A: 1,
+      E: 1,
+      I: 1,
+      O: 1,
+      U: 1,
+      L: 1,
+      N: 1,
+      R: 1,
+      S: 1,
+      T: 1,
+      D: 2,
+      G: 2,
+      B: 3,
+      C: 3,
+      M: 3,
+      P: 3,
+      F: 4,
+      H: 4,
+      V: 4,
+      W: 4,
+      Y: 4,
+      K: 5,
+      J: 8,
+      X: 8,
+      Q: 10,
+      Z: 10,
+  },
+
+  score: function(word) {
+    var score = 0,
+        word = word.toUpperCase();
+
+    for (var i = 0; i < word.length; i++) {
+      score += this.scoreChart[word[i]];
+    }
+
+    if (word.length == 7) {
+      score += 50;
+    }
+
+    return score
+  },
+
+  highestScoreFrom: function(wordArray) {
+    var scoresWithWords = [],
+        tiedScoredWords = [],
+        wordScores = [],
+        maxScore = 0,
+        maxWord = "";
+
+    for ( var i = 0; i < wordArray.length; i++ ) {
+      var score = Scoring.score(wordArray[i]);
+      if ( score > maxScore ) {
+        maxScore = score;
+        maxWord = wordArray[i];
+      }
+    }
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+    for ( var j = 0; j < wordArray.length; j++ ) {
+      var scoreTwice = Scoring.score(wordArray[j]);
+      if (scoreTwice == maxScore) {
+        tiedScoredWords.push(maxWord);
+        tiedScoredWords.push(wordArray[j]);
+      }
+    }
+
+    if (tiedScoredWords.length > 0) {
+      var shortest = tiedScoredWords.reduce(function (a, b) { return a.length < b.length ? a : b; });
```

Comment:
Reduce is still a bit confusing to me. the (a,b) represents just the first two elements of the array, right? Then they're "math'ed" together (however you specify) to become a new first element. Then the next time the loop runs, a becomes the new "sum" and b becomes the next element, right?